src/mento.ts

```diff
@@ -326,4 +336,111 @@ export class Mento {
+    )
+    return exchanges[0]
+  }
+
+  /**
+   * Returns whether trading is enabled in the given mode for a given exchange id
+   * @param exchangeId the id of the exchange
+   * @param mode the trading mode
+   * @returns true if trading is enabled in the given mode, false otherwise
+   */
+  async isTradingEnabled(
+    exchangeId: string,
+    mode: TradingMode = TradingMode.BI_DIRECTIONAL
+  ): Promise<boolean> {
+    const exchange = await this.getExchangeById(exchangeId)
+    const biPoolManager = BiPoolManager__factory.connect(
+      exchange.providerAddr,
+      this.signerOrProvider
+    )
+
+    const [breakerBoxAddr, exchangeConfig] = await Promise.all([
+      biPoolManager.breakerBox(),
+      biPoolManager.getPoolExchange(exchangeId),
+    ])
+
+    const breakerBox = IBreakerBox__factory.connect(
+      breakerBoxAddr,
+      this.signerOrProvider
+    )
+    const currentMode = await breakerBox.getRateFeedTradingMode(
+      exchangeConfig.config.referenceRateFeedID
+    )
+
+    return currentMode.toNumber() == mode
+  }
+
+  /**
+   * Return the trading limits for a given exchange id. Each limit is an object with the following fields:
+   * asset: the address of the asset with the limit
+   * maxIn: the maximum amount of the asset that can be sold
+   * maxOut: the maximum amount of the asset that can be bought
+   * until: the timestamp until which the limit is valid
+   * @param exchangeId the id of the exchange
+   * @returns the list of trading limits
+   */
+  async getTradingLimits(exchangeId: string): Promise<TradingLimit[]> {
+    const exchange = await this.getExchangeById(exchangeId)
+    const broker = Broker__factory.connect(
+      this.broker.address,
+      this.signerOrProvider
+    )
+
+    const assetWithLimit = exchange.assets[0] // currently limits are configured only on asset0
+    return getLimits(broker, exchangeId, assetWithLimit)
+  }
+
+  /**
+   * Returns the trading limits configuration for a given exchange id
+   * @param exchangeId the id of the exchange
+   * @returns the trading limits configuration
+   */
+  async getTradingLimitConfig(
+    exchangeId: string
+  ): Promise<TradingLimitsConfig> {
+    const exchange = await this.getExchangeById(exchangeId)
+    const broker = Broker__factory.connect(
+      this.broker.address,
+      this.signerOrProvider
+    )
+
+    const assetWithLimit = exchange.assets[0] // currently limits are configured only on asset0
+    return getLimitsConfig(broker, exchangeId, assetWithLimit)
+  }
+
+  /**
+   * Returns the trading limits state for a given exchange id
+   * @param exchangeId the id of the exchange
+   * @returns the trading limits state
+   */
+  async getTradingLimitState(exchangeId: string): Promise<TradingLimitsState> {
+    const exchange = await this.getExchangeById(exchangeId)
+    const broker = Broker__factory.connect(
+      this.broker.address,
+      this.signerOrProvider
+    )
```

Comment:
Seems like we do this a few times, maybe a `getBrokerContract` method would be useful

Reply:
Agree, however we already have one that returns the `IBroker` instance https://github.com/mento-protocol/mento-sdk/blob/61c2198b236d36e898a10f4acff5eb2ae72ffd55/src/mento.ts#L267-L269. I think it would make sense to change that one to return a `Broker` instead (since the interface is missing some methods that are needed elsewhere, which is why I couldn't re-use it). 

I'm fine with merging it as it's and changing it later on